ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -3407,16 +3407,21 @@ signal_list_t* create_operation_node(ast_node_t* op, signal_list_t** input_lists
             break;
         case SR:  // >>
         case ASR: // >>>
-            /* Shifts doesn't matter about port size, but second input needs to be a number */
-            output_port_width = input_lists[0]->count;
-            input_port_width = input_lists[0]->count;
-            break;
         case ASL: // <<<
         case SL:  // <<
-            /* Shifts doesn't matter about port size, but second input needs to be a number */
-            //output_port_width = input_lists[0]->count + (shift_left_value_with_overflow_check(0x1, input_lists[1]->count)-1);
-            output_port_width = input_lists[0]->count + (shift_left_value_with_overflow_check(0x1, log2(op->children[1]->types.vnumber->get_value()), op->loc));
-            input_port_width = output_port_width;
+            output_port_width = op->types.operation.output_port_width;
+            input_port_width = (input_lists[0]->count > input_lists[1]->count) ? input_lists[0]->count : input_lists[1]->count;
+
+            if (output_port_width > input_port_width)
+                input_port_width = output_port_width;
+            /*
+             * This condition checks the output_port_width has been resolved in ast phase if it hasn't 
+             * that means the there was no LHS operand size or in another word it wasnot a statement! 
+             * As a result, its value would be set to the first RHS operand. 
+             * e.x. array[(addr/4)*2] => (addr/4)*2 => output_port_width = addr_port_width
```

Comment:
this actually does also have an assignment size, which would be the log2( of the address space ). 

This is a bigger problem I think, good catch!